src/js/components/TextInput/StyledTextInput.js

```diff
@@ -23,11 +23,22 @@ const getPlainStyle = (plain) => {
 const StyledTextInput = styled.input`
   ${inputStyle}
   ${(props) => getPlainStyle(props.plain)}
-  ${(props) =>
-    props.icon &&
-    (props.reverse
-      ? `padding-right: ${props.theme.global.edgeSize.large};`
-      : `padding-left: ${props.theme.global.edgeSize.large};`)}
+  ${(props) => {
+    if (props.icon) {
+      const iconSize = props.theme.icon?.size?.[props?.size || 'medium'];
+      const pad = props.theme.icon.matchSize
```

Comment:
This is a bit convoluted to follow. I'd suggest
- a comment explaining what's going on
- putting this in a function that can be shared with MaskedInput to avoid the duplication.